finesse/gui/stepper_motor_view.py

```diff
@@ -51,7 +63,25 @@ def _add_checkable_button(self, name: str) -> QPushButton:
     def _preset_clicked(self, btn: QPushButton) -> None:
         """Move the stepper motor to preset position."""
         # If the motor is already moving, stop it now
+        pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.notify_on_stopped")
         pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.stop")
 
         target = float(self.angle.value()) if btn is self.goto else btn.text().lower()
         pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.move.begin", target=target)
```

Comment:
The problem here is that the `move.end` message will be sent in response to the `stop` message, not the `move.begin` one below, which means it'll send the *last* angle as an argument. So:

```suggestion
        pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.stop")

        pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.notify_on_stopped")
        target = float(self.angle.value()) if btn is self.goto else btn.text().lower()
        pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.move.begin", target=target)
```